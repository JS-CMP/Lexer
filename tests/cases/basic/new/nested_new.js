function Foo() {
    return function () {
        console.log("Inside Foo")
        return "Hello, World!"
    }
}
var x = 123

new new Foo() // NEW(NEW(Foo))
new (new Foo)(x) // NEW(NEW(Foo), x)
console.log(new new Foo(x)(x)) // NEW(NEW(Foo, x), x)
