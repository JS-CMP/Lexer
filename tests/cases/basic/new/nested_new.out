#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {JS::Any Foo, x;Foo = JS::Any(JS::Function([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {return JS::Any(JS::Function([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {console["log"](JS::Any("Inside Foo"));return JS::Any("Hello, World!");return JS::Any();}));return JS::Any();}));x=JS::Any(123);NEW(NEW(Foo));NEW((NEW(Foo)),x);console["log"](NEW(NEW(Foo,x),x));}
